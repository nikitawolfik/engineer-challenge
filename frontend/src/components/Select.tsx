import { ChangeEvent } from "react";
import { InputProps } from "./Search";

type Option<T> = [T | "", string];

interface Props<T> extends Omit<InputProps, "onChange"> {
  options: Option<T>[];
  onChange: (
    e: ChangeEvent<HTMLSelectElement> & { target: { value: T | "" } }
  ) => void;
}

function Select<T extends string>({
  name,
  label,
  value,
  onChange,
  options,
}: Props<T>) {
  return (
    <div className="flex flex-col mb-2 flex-1">
      <label className="text-xs ml-2 mb-1" htmlFor={`select-${name}`}>
        {label}
      </label>
      <select
        className="rounded-lg p-2 border-solid border-slate-50 border-2"
        name={name}
        value={value}
        placeholder={label}
        onChange={onChange}
      >
        {options.map(([key, display]) => (
          <option key={key} value={key}>
            {display}
          </option>
        ))}
      </select>
    </div>
  );
}

export default Select;
