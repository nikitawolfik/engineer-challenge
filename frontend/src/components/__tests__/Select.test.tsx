/**
 * @jest-environment jsdom
 */

import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Select from "components/Select";

describe("Search input", () => {
  test("Snapshot", async () => {
    const { container } = render(
      <Select
        label="Hello label"
        name="Test name"
        value=""
        options={[
          ["one", "one"],
          ["two", "two"],
        ]}
        onChange={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
