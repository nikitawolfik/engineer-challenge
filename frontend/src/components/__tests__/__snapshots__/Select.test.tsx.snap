// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search input Snapshot 1`] = `
<div>
  <div
    class="flex flex-col mb-2 flex-1"
  >
    <label
      class="text-xs ml-2 mb-1"
      for="select-Test name"
    >
      Hello label
    </label>
    <select
      class="rounded-lg p-2 border-solid border-slate-50 border-2"
      name="Test name"
      placeholder="Hello label"
    >
      <option
        value="one"
      >
        one
      </option>
      <option
        value="two"
      >
        two
      </option>
    </select>
  </div>
</div>
`;
